street_network/src/lanes/osm2lanes.rs

```diff
@@ -0,0 +1,355 @@
+use anyhow::Result;
```

Comment:
This is adapted from https://github.com/a-b-street/abstreet/pull/890. I wrote it in April and have mostly lost context on some of the subtler choices. There's lot in here that's weird/wrong or is stuff that we should fix in osm2lanes proper. But it's a start!